### PR TITLE
Add AUTHORS

### PR DESCRIPTION
**Pull request by alexeykuzmin**
There are lots of duplicates in the list. Patches are welcome!

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -0,0 +1,18 @@
+=^._.^= <max@maxogden.com> <maxogden@users.noreply.github.com>
+Alexey Kuzmin <alex.s.kuzmin@gmail.com> <alexeykuzmin@users.noreply.github.com>
+Alexey Kuzmin <alex.s.kuzmin@gmail.com> <alkuzmin@microsoft.com>
+Aleš Pergl <alpergl@microsoft.com> <alespergl@users.noreply.github.com>
+Amos Wenger <fasterthanlime@gmail.com> <fasterthanlime@users.noreply.github.com>
+Destan Sarpkaya <destan@dorukdestan.com> <destan@users.noreply.github.com>
+Gabriel Handford <gabrielh@gmail.com>
+Ivan Mir <ivmirx@fastmail.fm> <ivmirx@users.noreply.github.com>
+Jacob Groundwater <groundwater@gmail.com> <groundwater@github.com>
```
Comment (alexeykuzmin) [inline .mailmap]: @groundwater, are you ok having your gmail address as a canonical one?

File: .mailmap [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+João Moreno <mail@joaomoreno.com> <joaomoreno@users.noreply.github.com>
+Machisté N. Quintana <mnquintana@users.noreply.github.com>
+Samuel Attard <samuel.r.attard@gmail.com> <samuel.attard@orionhealth.com>
+Samuel Attard <samuel.r.attard@gmail.com> <sattard@atlassian.com>
```
Comment (alexeykuzmin) [inline .mailmap]: @MarshallOfSound, are you ok having your gmail address as a canonical one?

File: .mailmap [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+Sergey Rubanov <chi187@gmail.com>
+Shelley Vohr <shelley.vohr@gmail.com> <codebytere@github.com>
```
Comment (alexeykuzmin) [inline .mailmap]: @MarshallOfSound, are you ok having your gmail address as a canonical one?

File: AUTHORS [hunk 4]
```diff
@@ -0,0 +1,935 @@
...
+Austin Burdine <acburdine@users.noreply.github.com>
+Austin Moore <austin_moore@me.com>
+BJR Matos <bjrmatos@gmail.com>
+Bastien <lisurc@users.noreply.github.com>
+Belleve Invis <belleve@typeof.net>
+Ben Drucker <bvdrucker@gmail.com>
+Ben Gotow <bengotow@gmail.com>
+Ben Randall <veleek@gmail.com>
+Ben Vinson <vinsonb@gmail.com>
+Benjamin Pasero <benjamin.pasero@gmail.com>
```
Comment (bpasero) [inline AUTHORS]: @alexeykuzmin I would prefer my mail below (from Microsoft)

File: AUTHORS [hunk 5]
```diff
@@ -0,0 +1,935 @@
...
+Eric Brumer <ericbr@microsoft.com>
+Eric Poulsen <eric@zyxod.com>
+Everardo Medina <emedina@rubixware.com>
+Evlogiy Sutormin <evlogij@gmail.com>
+FWeinb <Fabrice@weinberg.me>
+Fabian Zeindl <fabian.zeindl@gmail.com>
+Federico Bond <federicobond@gmail.com>
+Felipe Serafim <feserafim@gmail.com>
+Felix Rieseberg <felix@felixrieseberg.com>
+Felix Rieseberg <felixr@slack-corp.com>
```
Comment (felixrieseberg) [inline AUTHORS]: Let's cut the `@slack-corp` one ✂️

File: AUTHORS (hunk 4 above)
Comment (ckerr) [reply AUTHORS]: Fixed https://github.com/electron/electron/pull/13255/commits/2e55ffa2bb6aa828c9c60562448924b8adc7b583

File: AUTHORS [hunk 6]
```diff
@@ -0,0 +1,935 @@
...
+Jimmy Moon <ragingwind@gmail.com>
+JinHyeok Yeon <duswlsgur@daum.net>
+Joao Moreno <mail@joaomoreno.com>
+Joe Naha <joe.naha@gmail.com>
+Johan Giraldo <johanpalomino59@gmail.com>
+Johannes Schmitz <j13z@users.noreply.github.com>
+John Eismeier <32205350+jeis2497052@users.noreply.github.com>
+John Haley <john@haley.io>
+John Kleinschmidt <jkleinsc@github.com>
+John Kleinschmidt <kleinschmidtorama@gmail.com>
```
Comment (ckerr) [inline AUTHORS]: @jkleinsc which email address would you prefer as your canonical one?

File: AUTHORS [hunk 7]
```diff
@@ -0,0 +1,935 @@
...
+Yoyo Zhou <yozhov@gmail.com>
+Yudu Ban <banyudu@gmail.com>
+Yui <yuit@users.noreply.github.com>
+Yury Solovyov <biohazard707@gmail.com>
+Yuya Ochiai <yuya0321@gmail.com>
+Zachary Flower <zach@ninjaninja.net>
+Zachary Flower <zach@zacharyflower.com>
+Zeke "Offline in April" Sikelianos <zeke@github.com>
+Zeke Sikelianos <zeke@github.com>
+Zeke Sikelianos <zeke@sikelianos.com>
```
Comment (ckerr) [inline AUTHORS]: @zeke which of these three emails would you like to use as your canonical one?

File: AUTHORS (hunk 6 above)
Comment (jkleinsc) [reply AUTHORS]: @ckerr the GitHub one

File: .mailmap (hunk 3 above)
Comment (codebytere) [reply .mailmap]: lord how do i have this many emails

Comment (codebytere) [reply .mailmap]: csvohr1995@gmail.com, yeah

File: AUTHORS [hunk 8]
```diff
@@ -0,0 +1,935 @@
...
+Michael J. Zoidl <me@michaelzoidl.com>
+Michael Kaylan <1063516+kaylanm@users.noreply.github.com>
+Michael Köppl <mko.mko31@gmail.com>
+Michael S. Barthelemy <msb@idiom.com>
+Michael Vasseur <michael.vasseur@thomsonreuters.com>
+Michael Vasseur <michael_vasseur@outlook.com>
+Miguel Piedrafita <github@miguelpiedrafita.com>
+Mikael Finstad <finstaden@gmail.com>
+Mike Engel <mike@mike-engel.com>
+Milan Burda <miburda@microsoft.com>
```
Comment (miniak) [inline AUTHORS]: @alexeykuzmin Please list my gmail address first

**Comment by miniak** [comment]
@alexeykuzmin what’s the point if you can generate the list from git anytime? Who is going to make sure that it’s kept up to date?

**Comment by alexeykuzmin** [comment]
@miniak 

> Who is going to make sure that it’s kept up to date?

Someone who cares.
I don't see reasons why is should be automated.

**Comment by tonyganch** [comment]
> Someone who cares.

I don't think that adding some code to the repo with the thought that nobody will maintain it in the future is the right approach.

**Comment by alexeykuzmin** [comment]
@MarshallOfSound 
> What was this aiming to solve that the "Contributors" tab doesn't?

"[Contributors](https://github.com/electron/electron/graphs/contributors)" tab lists only a 100 most active contributors while Electron has got contributions from 700+ people. And it's a ranked list, which means that a first-time contributor probably doesn't have a chance to get there.

**Comment by alexeykuzmin** [comment]
> What is the purpose of this PR?

To explicitly list everyone who helped Electron to get better.
The idea is stolen without a twinge of conscience from [Chromium](https://chromium.googlesource.com/chromium/src.git/+/master/AUTHORS), [Node.js](https://github.com/nodejs/node/blob/master/AUTHORS), and [React](https://github.com/facebook/react/blob/master/AUTHORS).

File: .mailmap (hunk 3 above)
Comment (alexeykuzmin) [reply .mailmap]: @codebytere Done.

File: AUTHORS (hunk 6 above)
Comment (alexeykuzmin) [reply AUTHORS]: @jkleinsc Done

File: AUTHORS (hunk 8 above)
Comment (alexeykuzmin) [reply AUTHORS]: @miniak Done

**Comment by alexeykuzmin** [comment]
@miniak 

> Who is going to make sure that it’s kept up to date?

We can explicitly ask all first-time contributors to add themselves to the AUTHORS list.

**Comment by alexeykuzmin** [comment]
> Syed Ahsan Ahmed snahmed1998@gmail.com is missing

@AhsanSN
Can you please give a link to any commit in this repository made by you?
You probably used a different email address and a name. Thanks!

File: script/authors [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+
+# Creates a list of all authors in the alphabetical order respecting .mailmap.
```
Comment (zeke) [inline script/authors]: What is `.mailmap.`?

File: AUTHORS (hunk 7 above)
Comment (zeke) [reply AUTHORS]: `Zeke Sikelianos <zeke@sikelianos.com>` :)

**Comment by zeke** [comment]
> Who is going to make sure that it’s kept up to date?

Could just be part of the build process.

**Comment by tonyganch** [comment]
Still -1 from me. I don't see any real value in those files, and "other projects do that" is not a good enough argument.

File: script/authors (hunk 9 above)
Comment (alexeykuzmin) [reply script/authors]: @zeke 
Good question ) Here's a description: https://git-scm.com/docs/git-check-mailmap#_mapping_authors
Also added a comment in the script with the same link.

**Comment by alexeykuzmin** [comment]
@codebytere 
If the Electron's team doesn't see any value in having it then yes, of course it can be closed.

**Pull request closed without merging**